R/guides-axis.r

```diff
@@ -1,4 +1,157 @@
 
+#' Axis guide
+#'
+#' Axis guides are the visual representation of position scales like those
+#' created with [scale_(x|y)_continuous()][scale_x_continuous()] and
+#' [scale_(x|y)_discrete()][scale_x_discrete()].
+#'
+#' @inheritParams guide_legend
+#' @param check.overlap silently remove overlapping labels,
+#'   (recursively) prioritizing the first, last, and middle labels.
+#' @param angle Compared to setting the angle in [theme()] / [element_text()],
+#'   this also uses some heuristics to automatically pick the `hjust` and `vjust` that
+#'   you probably want.
+#' @param n_dodge The number of rows (for vertical axes) or columns (for
+#'   horizontal axes) that should be used to render the labels. This is
+#'   useful for displaying labels that would otherwise overlap.
+#' @param order Used to determine the order of the guides (left-to-right,
+#'   top-to-bottom), if more than one  guide must be drawn at the same location.
+#' @param position Where this guide should be drawn: one of top, bottom,
+#'   left, or right.
+#'
+#' @export
+#'
+#' @examples
+#' # plot with overlapping text
+#' p <- ggplot(mpg, aes(cty * 100, hwy * 100)) +
+#'   geom_point() +
+#'   facet_wrap(vars(class))
+#'
+#' # axis guides can be customized in the scale_* functions or
+#' # using guides()
+#' p + scale_x_continuous(guide = guide_axis(n_dodge = 2))
+#' p + guides(x = guide_axis(angle = 90))
+#'
+#' # can also be used to add a duplicate guide
+#' p + guides(x = guide_axis(n_dodge = 2), y.sec = guide_axis())
+#'
+#'
+guide_axis <- function(title = waiver(), check.overlap = FALSE, angle = NULL, n_dodge = 1,
```

Comment:
I think we should stick with old ggplot2 style and call it `n.dodge` instead of `n_dodge` (though I don't like it 🙂)